### PR TITLE
Bug 590643

### PR DESCRIPTION
**Pull request by kaadhina**

### CODE REVIEW

**Comment by kaadhina** [comment]
Is the fix already in master? In progress.
What testing was done? Manual
Does this fix change the AT/DT contract? No
Were tests added to cover this to make sure it doesn't regress? Will add in master
Is this a regression? No

Will make additional dependency change in mseng after this

**Comment by acesiddhu** [comment]
Can u add testing done?

File: Tasks/VsTest/Helpers.ps1 [hunk 1]
```diff
@@ -43,6 +43,33 @@ function IsVisualStudio2015Update1OrHigherInstalled {
...
+		if($resultsDirElement -And $resultsDirElement.Count -ne 0)
+		{
+            $customLocation = $runSettingsForTestResults.RunSettings.RunConfiguration.ResultsDirectory       
+            if([io.path]::IsPathRooted($customLocation))
+            {
+                return $customLocation
+            }
+            else
+            {
+                return [io.path]::GetFullPath([io.path]::Combine([io.path]::GetDirectoryName($runSettingsFilePath), $customLocation))
```
Comment (acesiddhu) [inline Tasks/VsTest/Helpers.ps1]: can u check resetting parsing in rocksteady is also doing the same thing?

**Comment by kaadhina** [comment]
Testing done: Manual validations of
Absolute path, relative path, 
inaccessible path (will throw inaccessible error)
With and without run parameters

**Comment by kaadhina** [comment]
With run parameters: will not publish results. Customers will have to continue to use the current workaround.

**Comment by acesiddhu** [comment]
Test cases
1) Normal scenarios continue to work
2) Scenarios with Relative runsettings will now start working
3) Scenarios where folks have existing build definition with relative path with workaround apply whats the behavior?
4) Scenarios with Absolute runsettings will now start working
5) Run all above two times back to back. Want to see if there are stale binaries left.

**Comment by kaadhina** [comment]
All test cases covered.
# 3 - It will publish twice until u remove the publish task in definition

File: Tasks/VsTest/Helpers.ps1 (hunk 1 above)
Comment (acesiddhu) [inline Tasks/VsTest/Helpers.ps1]: I see that when run in parallel is enabled and user is using relative path we wont work. is it true?

**Comment by acesiddhu** [comment]
6) test with and without relative path and run in parallel checked

**Comment by kaadhina** [comment]
Yes, if parallel is check and relative path is given we wont publish. Only absolute path will work.

File: Tasks/VsTest/Helpers.ps1 (hunk 1 above)
Comment (acesiddhu) [inline Tasks/VsTest/Helpers.ps1]: shuld we check before returning if the path  exists?

Comment (acesiddhu) [inline Tasks/VsTest/Helpers.ps1]: make sure l0 test r passing

**Comment by acesiddhu** [comment]
One more test. Run with chutzpah

**Comment by acesiddhu** [comment]
buddy tested
1) With parallel and relative path (works)
2) With no param and relative path (works)
3) With param and relative path (does not work)
4) With param and with workaround (**fails**)
5) Without param and with workaround (publishes **two** times result)

**Comment by acesiddhu** [comment]
6) Normal run and parallel run without runsetting works
7) Normal run and parallel run with normal runsetting works

**Comment by acesiddhu** [comment]
looks fine to me now

**Comment by acesiddhu** [comment]
all above testing with absolute oath as well. Kavipriya has fixed #4 above now. signing off on testing as well

**Pull request merged**